src/flow.ts

```diff
@@ -77,167 +77,214 @@ import { Vibrato } from './vibrato';
+  static TickContext = TickContext;
+  static TimeSignature = TimeSignature;
+  static TimeSigNote = TimeSigNote;
+  static Tremolo = Tremolo;
+  static Tuning = Tuning;
+  static Tuplet = Tuplet;
+  static Vibrato = Vibrato;
+  static VibratoBracket = VibratoBracket;
+  static Voice = Voice;
+  static Volta = Volta;
 
-  // BUILD and VERSION are set by webpack. See: Gruntfile.js.
-  BUILD: '',
-  VERSION: '',
+  // enums need to be exported too!
+  static BarlineType = BarlineType;
+  static ModifierPosition = ModifierPosition;
+  // ... more to come ...
 
-  get DEFAULT_FONT_STACK(): Font[] {
-    return Tables.DEFAULT_FONT_STACK;
-  },
-  set DEFAULT_FONT_STACK(value: Font[]) {
-    Tables.DEFAULT_FONT_STACK = value;
-  },
-  get DEFAULT_NOTATION_FONT_SCALE(): number {
-    return Tables.DEFAULT_NOTATION_FONT_SCALE;
-  },
-  set DEFAULT_NOTATION_FONT_SCALE(value: number) {
-    Tables.DEFAULT_NOTATION_FONT_SCALE = value;
-  },
-  get DEFAULT_TABLATURE_FONT_SCALE(): number {
-    return Tables.DEFAULT_TABLATURE_FONT_SCALE;
-  },
-  set DEFAULT_TABLATURE_FONT_SCALE(value: number) {
-    Tables.DEFAULT_TABLATURE_FONT_SCALE = value;
-  },
-  get RESOLUTION(): number {
+  // VERSION and BUILD are set by webpack string-replace-loader. See: Gruntfile.js.
+  static VERSION: string = '_VEX_VERSION_';
+  static BUILD: string = '_VEX_BUILD_';
+
+  /**
+   * `Flow.setMusicFont(...fontNames)` behaves differently depending on how you use VexFlow.
+   *
+   * **CASE 1**: You are using `vexflow.js`, which includes all music fonts (Bravura, Gonville, Petaluma, Custom).
+   * In this case, calling this method is optional, since VexFlow already defaults to a music font stack of:
+   * 'Bravura', 'Gonville', 'Custom'. This method is synchronous.
+   *
+   * Examples:
+   * ```
+   * Vex.Flow.setMusicFont('Petaluma');
+   * Vex.Flow.setMusicFont('Bravura', 'Gonville');
+   * ```
+   *
+   * **CASE 2**: You are using the lighter weight `vexflow-core.js` to take advantage of lazy loading for fonts.
+   * In this case, you MUST call this method at the beginning, since the default music font stack is empty.
+   * This method is replaced by an async function, so you must use `await` or a Promise to wait for the fonts
+   * to load before proceeding. See `demos/fonts/` for examples. See `src/fonts/async.ts` for implementation details.
+   *
+   * Example:
+   * ```
+   * await Vex.Flow.setMusicFont('Petaluma');
+   * ... (do VexFlow stuff) ...
+   * ```
+   * @returns CASE 1: an array of Font objects corresponding to the provided `fontNames`.
+   * @returns CASE 2: Promise<Font[]> that resolves to the same array of Font objects as above.
+   */
+  static setMusicFont(...fontNames: string[]): Font[] {
+    // Convert the array of font names into an array of Font objects.
+    const fonts = fontNames.map((fontName) => Font.load(fontName));
+    Tables.MUSIC_FONT_STACK = fonts;
+    Glyph.CURRENT_CACHE_KEY = fontNames.join(',');
+    return fonts;
```

Comment:
Music engraving fonts are now specified by string (e.g., 'Bravura', 'Petaluma'). Previously we were creating Font objects, and sometimes we'd create multiple Font.Bravura() objects.

This new approach is also more compatible with dynamic font loading. Once a font is dynamically loaded once, it is added to the cache, and future calls to Font.load(fontName) will return the same Font object (See: src/fonts/async.ts).